### PR TITLE
Add support for ignore-unknown-events flag

### PR DESCRIPTION
**Pull request by DougHamil**
Hello!

I noticed there was a `:ignore-unknown-event?` flag in the `fsm/transition` call but it wasn't being used to disable the error throw when the event is unknown.

This PR adds support for that flag.

Also, I wanted to expose that flag so that it can be used with the Service. I added an optional `:transition-opts` that is passed in when creating the Service and will be passed to the `fsm/transition` call by the Service. I'm not sure if this is ideal, though. Maybe I should instead add another arity to the `service/send` method so the opts can be passed there?

### CODE REVIEW

File: src/statecharts/impl.cljc [hunk 1]
```diff
@@ -734,11 +734,12 @@
...
                  (map #(select-one-tx fsm % state event input-event)))
-        _ (when-not (->> txs
+        _ (when-not ignore-unknown-event?
+            (when-not (->> txs
                          (map first)
                          (some identity))
-            (throw (ex-info (str "fsm " (:id fsm) " got unknown event " (:type event))
-                            {:_state _state})))
+              (throw (ex-info (str "fsm " (:id fsm) " got unknown event " (:type event))
+                              {:_state _state}))))
```
Comment (lucywang000) [inline src/statecharts/impl.cljc:742]: Hey @DougHamil thanks for this patch. The change should work, but with all due respect I don't like mixing the major changes with reformatting of code. 

So would you please either:
1. revert all formatting/whitespace changes, or
2. split the whitespace changes (that are not related to this patch) into a separate commit so each commit is atomic itself.

Comment (lucywang000) [reply src/statecharts/impl.cljc:742]: Also the CI is not running for this PR. I have updated the master branch to enable github actions on pull requests, so please rebase your branch to make the CI run on it.

Comment (DougHamil) [reply src/statecharts/impl.cljc:742]: Whoops, my parinfer got a little aggressive! I've reverted the whitespace changes and rebased on latest master.

Let me know if there are any other adjustments I should make!
